platforms/core-configuration/file-collections/src/integTest/groovy/org/gradle/api/file/FileCollectionIntegrationTest.groovy

```diff
@@ -38,6 +38,21 @@ class FileCollectionIntegrationTest extends AbstractIntegrationSpec implements T
         type << ["Object", "Object[]", "Set", "LinkedHashSet", "List", "LinkedList", "Collection", "FileCollection"]
     }
 
+    def "can set elements as convention of file collection"() {
+        buildFile """
+            def files = objects.fileCollection().convention('b')
+            def elements = files.elements
+
+            def name = 'a'
+            files.configure { it.from(name) }
```

Comment:
@bamboo Any pointers on the incantation needed to set an implicit receiver for actions specified as Groovy closures?

Reply:
`files` needs to be decorated. That happens when it's created via the `ObjectFactory` or a properly configured `Instantiator`.

Reply:
Thanks, but not sure I follow. Isn't `files` above already created via `ObjectFactory`?

Reply:
Sorry, I didn't look the test close enough. Gradle doesn't decorate `Transformer` taking methods with `Closure` ones. That only happens for `Action` taking methods.

Reply:
FTR: https://github.com/gradle/gradle/blob/f0e72fb78af3019816aeff29018b2c70b4548b05/platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/AbstractClassGenerator.java#L906

Reply:
Mystery solved, `ObjectFactory#fileCollection()` returns a non-decorated file collection: https://github.com/gradle/gradle/blob/9239189ef8d307b43b765f5b9637f509d8f5a436/platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFileCollectionFactory.java#L112-L111

Reply:
Thanks - so what we have is ok, right? Or should we try to get the ability of setting the file collection as an implicit receiver some other way?

Reply:
It seems this is the first method in `CFC` to require the capability, we should consider changing the way the collection is instantiated.